default.nix

```diff
@@ -15,18 +15,32 @@ rec {
     in
       pkg.offline_cache;
 
-  buildYarnPackageDeps = { name, packageJson, yarnLock, yarnNix ? null }:
+  buildYarnPackageDeps = { name, packageJson, yarnLock, yarnNix ? null, pkgConfig ? {} }:
     let
       yarnNix_ =
         if yarnNix == null then (generateYarnNix yarnLock) else yarnNix;
       offlineCache =
         loadOfflineCache yarnNix_;
+      moreBuildInputs = (pkgs.lib.flatten (builtins.map (key:
+        pkgConfig.${key} . buildInputs or []
+      ) (builtins.attrNames pkgConfig)));
+      postInstall = (builtins.map (key:
+        if (pkgConfig.${key} ? "postInstall") then
+          ''
+            test -e node_modules/${key} && (
```

Comment:
That's mostly so you can add some common configuration in `buildYarnPackage` that can be used as a default by everybody, and they just add special cases they need. It's similar to how `bundlerEnv` works, just that we can't do it on a per-derivation basis but have to rely on directory names.
In theory we could use something like this:

    for f in $(find -L -path '*/node_modules/${key}' -type d); do
      (cd "$f" && (${pkgConfig.${key}.postInstall}))
    done